File: python3/1408.py
class Solution:
    def stringMatching(self, w: List[str]) -> List[str]:
        def isSub(w, t, i, n):
            for j in range(n):
                if i != j and w[i] in w[j]:
                    return True
            return False

        res = []
        n = len(w)
        for i in range(n):
            if isSub(w, w[i], i, n):
                res.append(w[i])
        return res

File: java/1408.java
class Solution {
    public List<String> stringMatching(String[] w) {
        List<String> res = new ArrayList<>();
        int n = w.length;
        for (int i = 0; i < n; i++)
            if (isSub(w, w[i], i, n))
                res.add(w[i]);
        return res;
    }

    private boolean isSub(String w[], String t, int i, int n) {
        for (int j = 0; j < n; j++)
            if (i != j && w[j].contains(w[i]))
                return true;
        return false;
    }
}
